sdk/core/System.ClientModel/src/Pipeline/ClientRetryPolicy.cs

```diff
@@ -28,6 +28,8 @@ public class ClientRetryPolicy : PipelinePolicy
     private readonly int _maxRetries;
     private readonly TimeSpan _initialDelay;
 
+    internal int? ClientLoggingPolicyIndex { get; set; }
```

Comment:
alternatives:
- same approach of having an internal property but instead holding the logging policy instance or the singleton
- TODO - others?